command/config/generate/analyzers_input.go

```diff
@@ -21,27 +35,138 @@ func (o *Options) collectAnalyzerInput() error {
-			o.GoImportRoot, err = utils.GetSingleLineInput(msg, helpMsg)
-			if err != nil {
-				return err
+	err = o.extractRequiredAnalyzerMetaFields()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Checks if the field is present in the array containing list of `optional_required`
+// analyzer meta fields
+func isContains(requiredFieldsList []string, field string) bool {
+	for _, v := range requiredFieldsList {
+		if field == v {
+			return true
+		}
+	}
+	return false
+}
+
+// Uses the `survey` prompt API to gather user input and store in `Options` struct
+// `Options` struct is later used for config generation
+func (o *Options) inputAnalyzerMeta(requiredFieldsData map[string][]AnalyzerMetadata) error {
+	var err error
+	// Iterate over the map and fetch the input for the fields from the user
+	for analyzer, metaFields := range requiredFieldsData {
+		for i := 0; i < len(metaFields); i++ {
+			switch metaFields[i].Type {
+			case "boolean":
+				metaFields[i].UserInput = "true"
+				res, err := utils.ConfirmFromUser(metaFields[i].Title, metaFields[i].Description)
+				if err != nil {
+					return err
+				}
+				if !res {
+					metaFields[i].UserInput = "false"
+				}
+			case "enum":
+				metaFields[i].UserInput, err = utils.SelectFromOptions(metaFields[i].Title, metaFields[i].Description, metaFields[i].Options)
+				if err != nil {
+					return err
+				}
+			default:
+				metaFields[i].UserInput, err = utils.GetSingleLineInput(metaFields[i].Title, metaFields[i].Description)
+				if err != nil {
+					return err
+				}
 			}
 		}
-		// TODO: Remove this hard coding
-		if analyzer == "Java (beta)" {
-			msg := "Which Java version does the project use?"
-			helpMsg := "The version of Java runtime to use. OpenJDK versions 8 to 15 are supported."
-			supportedJavaVersions := []string{"8", "9", "10", "11", "12", "13", "14", "15"}
-			o.JavaVersion, err = utils.SelectFromOptions(msg, helpMsg, supportedJavaVersions)
+		requiredFieldsData[analyzer] = metaFields
+	}
+	o.AnalyzerMetaMap = requiredFieldsData
+	return nil
+}
+
+// Extracts the fields that are compulsary according to the meta schema and require input
+func populateMetaData(optionalFields []string, jsonParsed *gabs.Container) []AnalyzerMetadata {
+	requiredFieldsData := make([]AnalyzerMetadata, len(optionalFields))
+
+	// Iterate through the properties using the parsed json (jsonParsed) and extract the data of the
+	// required analyzer meta fields
+	count := 0
+	for key, child := range jsonParsed.Search("properties").ChildrenMap() {
+		if !isContains(optionalFields, key) {
+			continue
+		}
+		propertyJSON, err := gabs.ParseJSON(child.Bytes())
+		if err != nil {
+			log.Printf("Error occured while parsing analyzer meta property: %v\n", err)
+			continue
+		}
+		requiredFieldsData[count].FieldName = key
+		requiredFieldsData[count].Title = propertyJSON.Search("title").Data().(string)
```

Comment:
Instead of making the array with `len(optionalFields)`, make it with length 0, and then append the new `AnalyzerMetadata` instance to the array.